### ISSUE TITLE
 allennlp test-install failed

### ISSUE THREAD

**Issue by susht3** [allenai/allennlp#1943]
help me please!!

2018-10-24 10:22:22,083 - INFO - allennlp.common.registrable - instantiating registered subclass relu of <class 'allennlp.nn.activations.Activation'>
2018-10-24 10:22:22,084 - INFO - allennlp.common.registrable - instantiating registered subclass relu of <class 'allennlp.nn.activations.Activation'>
2018-10-24 10:22:22,085 - INFO - allennlp.common.registrable - instantiating registered subclass relu of <class 'allennlp.nn.activations.Activation'>
2018-10-24 10:22:22,085 - INFO - allennlp.common.registrable - instantiating registered subclass relu of <class 'allennlp.nn.activations.Activation'>
2018-10-24 10:22:22,214 - INFO - allennlp.commands.test_install - Changing directory to /data1/susht3/local/anaconda3.6/lib/python3.6/site-packages
2018-10-24 10:22:22,215 - INFO - allennlp.commands.test_install - Running tests at /data1/susht3/local/anaconda3.6/lib/python3.6/site-packages/allennlp
========= test session starts =========
platform linux -- Python 3.6.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /data1/susht3/local/anaconda3.6/lib/python3.6/site-packages, inifile:
plugins: flaky-3.4.0
collected 1021 items

allennlp/tests/commands/configure_test.py .
allennlp/tests/commands/dry_run_test.py ......
allennlp/tests/commands/elmo_test.py /data1/susht3/local/anaconda3.6/bin/allennlp: line 2: 24025 Segmentation fault      (core dumped) python -m allennlp.run "$@"

**Comment by matt-gardner** [comment]
Most times I've seen this it is because of conflicts in pre-compiled packages like numpy. You might want to start from a totally clean environment, without any other versions of these packages installed.

**Comment by susht3** [comment]
> Most times I've seen this it is because of conflicts in pre-compiled packages like numpy. You might want to start from a totally clean environment, without any other versions of these packages installed.

thanks a lot, now i install all packages from allennlp, by "pip install --upgrade --force-reinstall allennlp", but now i get:

/data1/susht3/local/anaconda3/envs/allennlp/lib/python3.6/site-packages/allennlp/modules/text_field_embedders/basic_text_field_embedder.py:122: DeprecationWarning: the token embedders for BasicTextFieldEmbedder should now be specified as a dict under the 'token_embedders' key, not as top-level key-value pairs
  warnings.warn(DeprecationWarning("the token embedders for BasicTextFieldEmbedder should now "

-- Docs: https://docs.pytest.org/en/latest/warnings.html
 3 failed, 848 passed, 7 skipped, 21 deselected, 1 xfailed, 247 warnings in 124.68 seconds

**Comment by susht3** [comment]
> Most times I've seen this it is because of conflicts in pre-compiled packages like numpy. You might want to start from a totally clean environment, without any other versions of these packages installed.

like this：
/data1/susht3/local/anaconda3/envs/allennlp/lib/python3.6/site-packages/msgpack_numpy.py:214: PendingDeprecationWarning: encoding is deprecated, Use raw=False instead.
  return _unpackb(packed, **kwargs)
/data1/susht3/local/anaconda3/envs/allennlp/lib/python3.6/site-packages/msgpack_numpy.py:214: PendingDeprecationWarning: encoding is deprecated, Use raw=False instead.
  return _unpackb(packed, **kwargs)
<unknown>:1: DeprecationWarning: invalid escape sequence \s

**Comment by nelson-liu** [comment]
Can you give the traceback of the test failures? The things you're posting are (deprecation)warnings, not the failures.

**Comment by susht3** [comment]
> Can you give the traceback of the test failures? The things you're posting are (deprecation)warnings, not the failures.

i have 3 errors about highway tests, and i find that i can not import squad_eval by "from allennlp.tools import squad_eval", i get ImportError: cannot import name 'squad_eval'

one of the test-intall errors is:
llennlp/tests/custom_extensions/alternating_highway_lstm_test.py:46: in forward_and_backward_outputs_match
    kernel_output, _ = kernel_model(packed_kernel_input)
torch/nn/modules/module.py:477: in __call__
    result = self.forward(*input, **kwargs)
allennlp/modules/alternating_highway_lstm.py:252: in forward
    memory_accumulator, dropout_weights, lengths_variable, gates)
allennlp/modules/alternating_highway_lstm.py:50: in forward
    is_training)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<built-in method highway_lstm_forward_cuda of CompiledLib object at 0x7ff04b1d3408>, 103, 311, 83, 8, 101, ...)
kwargs = {}

    @wraps(function)
    def safe_call(*args, **kwargs):
        args = tuple(ffi.cast(_torch_to_cffi.get(arg.type(), 'void') + '*', arg._cdata)
                     if isinstance(arg, torch.Tensor) or torch.is_storage(arg)
                     else arg
                     for arg in args)
        args = (function,) + args
>       result = torch._C._safe_call(*args, **kwargs)
E       TypeError: initializer for ctype 'struct THIntTensor *' must be a pointer to same type, not cdata 'struct THLongTensor *'
torch/utils/ffi/__init__.py:202: TypeError

**Comment by matt-gardner** [comment]
The highway LSTM errors are expected, and are nothing that you need to worry about.  The `squad_eval` error is not expected, and I don't see it when I run off of master.  This is something that was recently changed (the script moved location), so my initial guess is that something is off in how your pip install worked.  It's also likely that you don't need to worry about that error, either.  I'm going to close this for now; feel free to re-open it if you have more questions.

**Comment by tingkai-zhang** [comment]
Hi! a following up question here.
I installed allennlp via pip 2 months ago.
Since allennlp has added some new models to its master branch. How can i import these new models?
I tried 
```
pip install -U pip
pip install -U allennlp
```
It didn't work because i can't import something like `BertEmbedder`.
I know it is probably not the right way to upgrade. But what should i do to use allennlp's up-to-date features? Thank you!

Also, i tried `allennlp test-install`, it gives
4 failed, 1050 passed, 10 skipped, 21 deselected, 1 xfailed, 392 warnings in 136.33 seconds
